src/common/droid-util.c

```diff
@@ -2180,6 +2182,105 @@ static int droid_output_stream_audio_patch_update(pa_droid_stream *primary_strea
     return ret;
 }
 
+int input_stream_set_parameter(pa_droid_stream *s, const dm_config_port *device_port){
+    pa_droid_input_stream *input;
+    audio_devices_t device;
+    audio_source_t source;
+    char *parameters = NULL;
+    int ret = 0;
+    pa_droid_hw_module *hw_module;
+
+
+    hw_module = s->module;
+    input = s->input;
+
+     /* Input stream closed, no need for set parameters */
+    if (!input->stream)
+        goto done;
+
+    device = device_port->type;
+    source = input->audio_source;
+
+    parameters = pa_sprintf_malloc("%s=%d;%s=%u", AUDIO_PARAMETER_STREAM_ROUTING, (int32_t) device,
+                                                  AUDIO_PARAMETER_STREAM_INPUT_SOURCE, source);
+
+
+    pa_log_debug("input stream %p set_parameters(%s) %#010x ; %#010x",
+                 (void *) s, parameters, device, source);
+
+
+    pa_mutex_lock(hw_module->input_mutex);
+    ret = input->stream->common.set_parameters(&input->stream->common, parameters);
+    pa_mutex_unlock(hw_module->input_mutex);
+
+    if (ret < 0) {
+        if (ret == -ENOSYS)
+            pa_log_warn("input set_parameters(%s) not allowed while stream is active", parameters);
+        else
+            pa_log_warn("input set_parameters(%s) failed", parameters);
+    }
+
+    pa_xfree(parameters);
+
+done:
+    return ret;
+
+}
+
+int output_stream_set_parameter(pa_droid_stream *s, const dm_config_port *device_port){
+    pa_droid_output_stream *output;
+    pa_droid_stream *slave;
+    uint32_t idx;
+    char *parameters = NULL;
+    int ret = 0;
+    audio_devices_t device;
+    pa_droid_hw_module *hw_module;
+    int set_bt_sco = -1;
+
+    pa_assert(s);
+    pa_assert(s->output);
+    pa_assert(s->module);
+    pa_assert(s->module->output_mutex);
+
+    output = s->output;
+    hw_module = s->module;
+    device = device_port->type;
+
+    pa_mutex_lock(s->module->output_mutex);
+
+
+    parameters = pa_sprintf_malloc("%s=%u;", AUDIO_PARAMETER_STREAM_ROUTING, device);
+
+    /* Set BT_SCO parameter for Bluetooth voice/voip call routes. */
+
+    set_bt_sco = (device & (AUDIO_DEVICE_OUT_BLUETOOTH_SCO |
```

Comment:
Just noting this one down for potential BT regressions, since we had some over time already and would like to rule out any unexpected changes.

But looks pretty clean for now.